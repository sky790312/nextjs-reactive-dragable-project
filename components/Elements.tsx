import React from 'react'
import styled from 'styled-components'
import Element from '@/components/Element'
import { useSelector, shallowEqual } from 'react-redux'
import { RootState } from '@/store/index'
import { useRouter } from 'next/router'

const Elements: React.FC = React.memo(() => {
  const router = useRouter()
  const blocks = useSelector(
    (state: RootState) => state.block.blocks,
    shallowEqual,
  )
  const pageBlocks = blocks.filter(
    block => block.name.split(' ')[1].charAt(0) === router.query.id,
  )

  return (
    <ElementsWrapper>
      <h4>Elements</h4>
      {pageBlocks.map(block => (
        <Element key={block.id} block={block} />
      ))}
    </ElementsWrapper>
  )
})

export default Elements

const ElementsWrapper = styled.div``
