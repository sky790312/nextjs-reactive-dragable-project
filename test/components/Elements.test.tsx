import React from 'react'
import { render, withProviders } from '@/test/testUtils'
import Elements from '@/components/Elements'

describe('Elements component', () => {
  it('matches snapshot', () => {
    const { asFragment } = render(withProviders(<Elements />), {})
    expect(asFragment()).toMatchSnapshot()
  })
})
