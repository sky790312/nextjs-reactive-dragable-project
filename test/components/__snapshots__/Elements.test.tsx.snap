// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Elements component matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="Elements__ElementsWrapper-vrh229-0"
  >
    <h4>
      Elements
    </h4>
  </div>
</DocumentFragment>
`;
